test-suite/soak-tests/src/services.rs

```diff
@@ -59,14 +59,18 @@ impl Service for MainService {
 
     fn after_commit(&self, context: AfterCommitContext<'_>) {
         if let Some(broadcaster) = context.broadcaster() {
-            if let Err(e) = broadcaster.timestamp((), context.height()) {
-                log::error!(
-                    "[{}] Failed to broadcast transaction at height {}: {}",
-                    context.service_key(),
-                    context.height(),
-                    e
-                );
-            }
+            let height = context.height();
+            let service_key = context.service_key();
+            futures::executor::block_on(async move {
```

Comment:
Maybe, it could make sense to introduce a sync version of broadcaster, at least as a temporary measure? Fragments similar to this one are repeated throughout the codebase.

Reply:
It make sense, but I don't know how to implement both of `GenericCall<(), Output = Future>` and  `GenericCall<(), Output = Result>` for the same context type.